inc/ConfigFile.hpp

```diff
@@ -27,6 +27,7 @@ struct Location {
 	std::map<statusCode, std::string> errorPage;
 	bool allowedMethods[MethodCount];
 	std::map<statusCode, std::string> returns;
+	std::pair<bool, std::string> alias;
```

Comment:
I'm not sure why the alias is stored as a ``std::pair<bool, std::string>``.

If I'm right, the bool is used to check if an alias has been set. But can't we just check if the alias string is empty instead? If it's empty, that means no alias was specified.

Reply:
> If I'm right, the bool is used to check if an alias has been set.

Yes you're correct.
 
> But can't we just check if the alias string is empty instead?

I thought if you want to define as alias an empty string. For me it doesn't make much sense but at least nginx accepts it.

But I have to admit I just tested it in the configfile but not what it actually does. This is what I found out:
I defined a location "/hello/" with alias "" [empty string]. On server level there was no root set.
I made a request to "hello/". It added /etc/nginx/ in front, so the actual request was not "" but "/etc/nginx/". 
When I set a server level root it made no difference (which makes sense since these two directives are mutually exclusive).
I could access a file I created at /etc/nginx/index.html with the uri "/hello/index.html" because it was translated to /etc/nginx/index.html

It seems that it has to do with how nginx resolves relative paths: /etc/nginx is the --prefix path of my nginx (checked with nginx -V). If you have a relative path nginx starts at the prefix path. E.g. an alias which does not start with / cannot be an absolute path since it does not start at /. 
If I set as alias "whatever" (no slash) it will replace /hello/ with "whatever" and then append this to "/etc/nginx/". Also alias = "../" would point to /etc/ (since ../ moved one level up).

I think relative path resolution can get messy pretty fast. 
I suggest we only allow absolute paths, which means we would need to check at parsing if alias start with slash "/". In this case we could also drop the std::pair<bool, std::string> like you suggested.

Reply:
Wow, I didn't think about the possibility to also specify an empty string as an alias. 

Even though I have to admit that I explicitly check this case for the directive server_name and accept it (nginx says it's okay and I also tested root now. Root also accepts an empty string).

That's an interesting behaviour of nginx. 

I will definitely check if an alias starts with ``/`` and if not I would throw an exception that only absolute paths are excepted. I will also look into the behaviour when an empty string is specified for server_name and root.

Reply:
I think root will show the same behavior: if it's a relative path (so no / at start) it will append it to --prefix path

Reply:
Yup, that's probably true. I am going to test it

Reply:
Refactored std::pair into std::string alias.

I will open issue to track that we don't want to implement relative path resolution. Meaning parsing needs to make sure root and alias start with '/'

Reply:
Great. Parsing will check, if root and alias start with ``/``.